packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -863,14 +871,18 @@ watch(() => props.schema, (newSchema, oldSchema) => {
   }
   Object.assign(originalModel, JSON.parse(JSON.stringify(form.model)))
 
-  freeformName.value = getFreeFormName(form.model.name, experimentalFreeForms)
+  freeformName.value = !props.engine
+    ? getFreeFormName(form.model.name, experimentalFreeForms)
+    : props.engine === 'vfg'
+      ? undefined
+      : getFreeFormName(form.model.name, experimentalFreeForms) || 'CommonForm'
   sharedFormName.value = getSharedFormName(form.model.name)
 
   initFormModel()
 }, { immediate: true, deep: true })
 
 onBeforeMount(() => {
-  form.value = parseSchema(props.schema)
+  form.value = parseSchema(props.schema, undefined, undefined, props.engine as any)
```

Comment:
Would this work without `any` now?

```suggestion
  form.value = parseSchema(props.schema, undefined, undefined, props.engine)
```